tpm2/structures.go

```diff
@@ -438,20 +442,31 @@ func DecodeAttestationData(in []byte) (*AttestationData, error) {
 	// The spec specifies several other types of attestation data. We only need
 	// parsing of Certify attestation data for now. If you need support for
 	// other attestation types, add them here.
-	if ad.Type != TagAttestCertify {
-		return nil, fmt.Errorf("only Certify attestation structure is supported, got type 0x%x", ad.Type)
-	}
-	if ad.AttestedCertifyInfo, err = decodeCertifyInfo(buf); err != nil {
-		return nil, fmt.Errorf("decoding AttestedCertifyInfo: %v", err)
+	switch ad.Type {
+	case TagAttestCertify:
+		if ad.AttestedCertifyInfo, err = decodeCertifyInfo(buf); err != nil {
+			return nil, fmt.Errorf("decoding AttestedCertifyInfo: %v", err)
+		}
+	case TagAttestCreation:
+		if ad.AttestedCreationInfo, err = decodeCreationInfo(buf); err != nil {
+			return nil, fmt.Errorf("decoding AttestedCreationInfo: %v", err)
+		}
+	default:
+		return nil, fmt.Errorf("only Certify & Creation attestation structures are supported, got type 0x%x", ad.Type)
 	}
+
 	return &ad, nil
 }
 
 // Encode serializes an AttestationData structure in TPM wire format.
 func (ad AttestationData) Encode() ([]byte, error) {
-	if ad.Type != TagAttestCertify {
-		return nil, fmt.Errorf("only Certify attestation structure is supported, got type 0x%x", ad.Type)
+	switch ad.Type {
```

Comment:
Remove this and add default case to the other switch below

Reply:
Will do!